.github/ISSUE_TEMPLATE.md

```diff
@@ -0,0 +1,15 @@
+Remember, an issue is not the place to ask questions. You can use [Stack Overflow](http://stackoverflow.com/questions/tagged/cakephp) for that.
+
+Before you open an issue, please check if a similar issue already exists or has been closed before.
+
+### When reporting a bug, please be sure to include the following:
+- [ ] A descriptive title
+- [ ] What version of CakePHP you're using, and the platform(s) you're running it on
+- [ ] The behavior you expect to see, and the actual behavior
+- [ ] An *isolated* way to reproduce the behavior
+
+### When you open an issue for a feature request, please add as much detail as possible:
+- [ ] A descriptive title
+- [ ] A description of the problem you're trying to solve, including *why* you think this is a problem
+- [ ] An overview of the suggested solution
+- [ ] If the feature changes current behavior, reasons why your solution is better
```

Comment:
These sections look less like a template and more like instructions on how to file an issue. Trimming down what @ionas suggested to something like

```
#### Preamble
This is a (multiple allowed):
* [x] bug
* [ ] enhancement
* [ ] feature-discussion (RFC)

## DESCRIPTIVE TITLE HERE
CakePHP Version: FULL SEM-VER OR COMMIT ID, HERE.
Platform and Target: YOUR WEB-SERVER, DATABASE AND OTHER RELEVANT INFO AND HOW THE REQUEST IS BEING MADE, HERE.

### Expected Behavior
EXPLAIN WHAT IS TO BE EXPECTED, HERE.

### Actual Behavior
EXPLAIN WHAT IS ACTUALLY HAPPENING, HERE.
```

I feel that less is more and we can always expand the issue template in the future.